plugins/database/src/db_plugin.cpp

```diff
@@ -938,31 +939,20 @@ bool _userInRUserAuth( const char* userName, const char* zoneName, const char* a
     }
 }
 
-// =-=-=-=-=-=-=-
-/// @brief function which determines if a char is allowed in a zone name
-static bool allowed_zone_char( const char _c ) {
-    return ( !std::isalnum( _c ) &&
-             !( '.' == _c )      &&
-             !( '_' == _c ) );
-} // allowed_zone_char
-
-// =-=-=-=-=-=-=-
-/// @brief function for validating the name of a zone
-irods::error validate_zone_name(
-    std::string _zone_name ) {
-    std::string::iterator itr = std::find_if( _zone_name.begin(),
-                                _zone_name.end(),
-                                allowed_zone_char );
-    if ( itr != _zone_name.end() || _zone_name.length() >= NAME_LEN ) {
-        std::stringstream msg;
-        msg << "validate_zone_name failed for zone [";
-        msg << _zone_name;
-        msg << "]";
-        return ERROR( SYS_INVALID_INPUT_PARAM, msg.str() );
+static irods::error validate_zone_name(const std::string& _zone_name)
+{
+    // Verify the zone name satisfies the requirements defined by the configuration
+    // schema. It would be nice to get the regular expression from the schema file, but
+    // we can't do that because iRODS 4 supports remote schema files. However, hard-coding
+    // the regular expression is also fine because no one should modify the schema (even
+    // though it's modifiable).
```

Comment:
The comment on the regex is a bit confusing to me.

If the comment is to highlight potential future issues by hard-coding the regex, the comment should be simplified to that. State the assumption of a default iRODS 4 regex and that issues may arise if the regex file is changed (if I understood the comment).

Reply:
Will revisit wording.

Reply:
Removed wording in favor of TODO comment.

Issue coming soon.